maven/src/org/netbeans/modules/maven/customizer/CompilePanel.java

```diff
@@ -322,6 +322,10 @@ public void setValue(Union2<JavaPlatform,String> value) {
                 }
             }
         };
+        // the selected item is not set until the compile panel is shown
+        // so, invoke these methods for setting it
+        compleComboBoxUpdater.ancestorAdded(null);
+        compleComboBoxUpdater.ancestorRemoved(null);
```

Comment:
A question: why is it needed to set the value here? (I.e. can't it wait until the compile panel is shown as it seems was done before?)

Reply:
Thank you for reviewing it. Because the source level combo box model uses the value of platform one.
e.g. 
If JDK 10 is already set to project properties when JDK 8 and JDK 10 are set to the Platform manager, JDK 10 might not be selected in the platform combo box correctly yet. i.e. If project properties panel is opened then the source category is clicked soon, the source level combo box might not have "9" and "10".

Reply:
@jlahoda Is this OK?